src/stubgen.py

```diff
@@ -81,11 +87,49 @@ class and repeatedly call ``.put()`` to register modules or contents within the
 ]
 # fmt: on
 
+# (name, desired_as_name) -> actual_as_name
+ImportDict = Dict[Tuple[Optional[str], Optional[str]], Optional[str]]
+# package_name -> ((name, desired_as_name) -> actual_as_name)
+PackagesDict = Dict[str, ImportDict]
+
+# (signature_str, doc_str, default_arg_1, default_arg_2, ...)
+# We could type as Tuple[Optional[str], Optional[str], *Tuple[Any, ...]]
```

Comment:
That is neat but doesn't seem like a useful strategy for now because of the minumum Python requirement. Let's remove that tangential comment.